gnuradio-runtime/lib/math/random.cc

```diff
@@ -71,6 +76,28 @@ namespace gr {
     if(seed==0) d_seed = static_cast<unsigned int>(std::time(0));
     else d_seed = seed;
     d_rng->seed(d_seed);
+    // reinstantiate generators. Otherwise reseed doesn't take effect.
+    delete d_generator;
```

Comment:
@stwunsch and I had a look at reseed and the corresponding test. We realized it doesn't have an effect if the generators aren't recreated. If the generator pointer members were completely uninitialized they would cause a segFault in 'reseed'. But according to some sources calling delete on a NULL object can be done safely. I considered that to be the best option.